From bf5bb1a3a469902e055072a9f703120493c43bf0 Mon Sep 17 00:00:00 2001
From: Manish Dharanenthiran <quic_mdharane@quicinc.com>
Date: Thu, 19 May 2022 12:03:44 +0530
Subject: [PATCH] ath12k: Comment out RX_MPDU_END and RX_MSDU_END stats parsing

HAL_RX_MSDU_END and HAL_RX_MPDU_END stats will be updated after
adding support to MON BUF ADDR TLV(s). Hence, commenting out these
two TLV parsing to avoid any NULL pointer dereference.

Signed-off-by: Manish Dharanenthiran <quic_mdharane@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp_mon.c | 17 ++++-------------
 1 file changed, 4 insertions(+), 13 deletions(-)

--- a/drivers/net/wireless/ath/ath12k/dp_mon.c
+++ b/drivers/net/wireless/ath/ath12k/dp_mon.c
@@ -826,22 +826,15 @@ ath12k_dp_mon_rx_parse_status_tlv(struct
 		break;
 	}
 	case HAL_RX_MSDU_END: {
-		struct rx_msdu_end_qcn9274 *msdu_end =
-			(struct rx_msdu_end_qcn9274 *)tlv_data;
-		bool is_first_msdu_in_mpdu;
-		u16 msdu_end_info;
-
-		msdu_end_info = __le16_to_cpu(msdu_end->info5);
-		is_first_msdu_in_mpdu = u32_get_bits(msdu_end_info,
-						     RX_MSDU_END_INFO5_FIRST_MSDU);
-		if (is_first_msdu_in_mpdu) {
-			pmon->mon_mpdu->head = pmon->mon_mpdu->tail;
-			pmon->mon_mpdu->tail = NULL;
-		}
+		/* TODO: RX_MSDU TLV will be handled once MON BUF ADDR TLVs are
+		 * enabled
+		 */
 		break;
 	}
 	case HAL_RX_MPDU_END:
-		list_add_tail(&pmon->mon_mpdu->list, &pmon->dp_rx_mon_mpdu_list);
+		/* TODO: RX_MSDU TLV will be handled once MON BUF ADDR TLVs are
+		 * enabled
+		 */
 		break;
 	case HAL_DUMMY:
 		return HAL_RX_MON_STATUS_BUF_DONE;
